plugins/wasm-rust/src/plugin_wrapper.rs

```diff
@@ -69,26 +96,99 @@ pub trait HttpContextWrapper<PluginConfig>: HttpContext {
     fn on_http_response_complete_body(&mut self, _res_body: &Bytes) -> DataAction {
         DataAction::Continue
     }
+
+    #[allow(clippy::too_many_arguments)]
+    fn on_http_call_response_detail(
+        &mut self,
+        _token_id: u32,
+        _arg: HttpCallArg,
+        _status_code: u16,
+        _headers: &MultiMap<String, String>,
+        _body: Option<Vec<u8>>,
+    ) {
+    }
     fn replace_http_request_body(&mut self, body: &[u8]) {
         self.set_http_request_body(0, i32::MAX as usize, body)
     }
     fn replace_http_response_body(&mut self, body: &[u8]) {
         self.set_http_response_body(0, i32::MAX as usize, body)
     }
+
+    fn get_http_call_storage(&mut self) -> Option<&mut HttpCallArgStorage<HttpCallArg>> {
+        None
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn http_call(
+        &mut self,
+        cluster: &dyn Cluster,
+        method: &Method,
+        raw_url: &str,
+        headers: MultiMap<String, String>,
+        body: Option<&[u8]>,
+        arg: HttpCallArg,
+        timeout: Duration,
+    ) -> Result<u32, Status> {
+        if let Ok(uri) = raw_url.parse::<Uri>() {
+            let mut authority = cluster.host_name();
+            if let Some(host) = uri.host() {
+                authority = host.to_string();
+            }
+            let mut path = uri.path().to_string();
+            if let Some(query) = uri.query() {
+                path = format!("{}?{}", path, query);
+            }
+            let mut headers_vec = Vec::new();
+            for (k, v) in headers.iter() {
+                headers_vec.push((k.as_str(), v.as_str()));
+            }
+            headers_vec.push((":method", method.as_str()));
+            headers_vec.push((":path", &path));
+            headers_vec.push((":authority", &authority));
+            let ret = self.dispatch_http_call(
+                &cluster.cluster_name(),
+                headers_vec,
+                body,
+                Vec::new(),
+                timeout,
+            );
+
+            if let Ok(token_id) = ret {
+                if let Some(storage) = self.get_http_call_storage() {
+                    storage.set(token_id, arg);
+                    log(
```

Comment:
Not use raw log (hostcall::log) directly, but use wrapped Log in wasm-rust

Reply:
done